cpu/nrf52/radio/nrf802154/nrf802154.c

```diff
@@ -131,6 +131,22 @@ static void _enable_tx(void)
     NRF_RADIO->TASKS_START = 1;
  }
 
+/**
+ * @brief basic ieee802.15.4 data frame validation
+ */
+static bool _data_frame_filter(void)
+{
+    size_t expected_len = ieee802154_get_frame_hdr_len(&rxbuf[1]);
```

Comment:
Is it guaranteed, that `rxbuf` is of length 2 at least? Otherwise this needs to be checked as well.

Reply:
I think the code would end up interpreting the CRC as frame header. So yeah, I would have to add an additional check here.

Reply:
Added an additional check to validate the minimum psdu size first.